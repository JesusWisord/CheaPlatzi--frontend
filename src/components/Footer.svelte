<style>
  footer {
    background-color: var(--variant-white);
    color: #000;
    gap: normal;
    widows: 100%;
  }

  a {
    text-decoration: none;
  }

  img {
    width: 100px;
    height: 110px;
  }
  .footer__links {
    display: flex;
    gap: normal;
    justify-content: center;
    margin: 0px 40px;
    padding: 48px 24px 0px 0px;
  }

  .footer__navigation {
    gap: normal;
    list-style: none;
    text-align: center;
  }

  .footer__links--title {
    display: list-item;
    font-size: 14px;
    gap: normal;
    line-height: 20px;
    margin: 0px 0px 10px;
    text-align: left;
  }

  .footer__logotype {
    gap: normal;
    margin: 0px 40px;
    justify-content: center;
    padding: 48px 24px 0px;
    text-align: center;
  }

  .footer__wrapper {
    align-items: center;
    display: flex;
    gap: normal;
    justify-content: center;
    margin: 0px 40px;
    padding: 24px 24px 48px;
  }

  .footer__copyright {
    font-size: 14px;
    gap: normal;
  }

  .footer__social {
    display: inline-flex;
    gap: normal;
  }

  @media (max-width: 375px) {
    .footer__wrapper {
      display: block;
      text-align: center;
    }
    .footer__social {
      color: var(--lightpurple-text);
      padding: 0;
    }
  }
</style>

<footer>

  <div class="footer__logotype">
    <img src="../gamecheap-logo.png" alt="Gamecheap Logo" />
    <h2>Gamecheap</h2>
  </div>

  <div class="footer__links">
    <ul class="footer__navigation">
      <li class="footer__links--title">
        <p>Company</p>
      </li>
      <li>
        <a href=".">Home</a>
      </li>
      <li>
        <a href=".">About</a>
      </li>
      <li>
        <a href=".">Blog</a>
      </li>
    </ul>

    <ul class="footer__navigation">
      <li class="footer__links--title">
        <p>Product</p>
      </li>
      <li>
        <a href=".">Info</a>
      </li>
      <li>
        <a href=".">Info</a>
      </li>
      <li>
        <a href=".">Info</a>
      </li>
    </ul>
  </div>

  <div class="footer__wrapper">
    <span class="footer__copyright">
      <p>Copyright 2020 Gamecheap. All rights reserved.</p>
    </span>
    <ul class="footer__social">
      <i class="github" />
      <a href=".">freshxcrapingenchiladas@gmail.com</a>
    </ul>
  </div>

</footer>
