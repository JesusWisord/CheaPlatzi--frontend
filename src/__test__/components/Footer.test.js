import { render as stlRender } from '@testing-library/svelte';
import '@testing-library/jest-dom/extend-expect';
import Footer from '../../components/Footer.svelte';

describe('<Footer /> Component', () => {
  test('Should render component correctly', () => {
    const { container } = stlRender(Footer);

    expect(container).toContainHTML("<body><div><footer><h2>Gamecheap 2020</h2></footer></div></body>")
  })
})

describe('Testing heading', () => {
  test('Shows proper heading when rendered', () => {
    const { getByText } = stlRender(Footer, { name: 'Gamecheap 2020' })
    expect(getByText('Gamecheap 2020')).toBeInTheDocument()
  })
})

describe('Footer Snapshot', () => {
  test('Check the UI of the Footer component', () => {
    const footer = stlRender(Footer)

    expect(footer).toMatchSnapshot()
  })
})