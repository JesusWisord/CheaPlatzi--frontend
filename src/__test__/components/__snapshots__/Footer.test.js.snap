// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Snapshot Check the UI of the Footer component 1`] = `
Object {
  "component": Footer {
    "$$": Object {
      "after_update": Array [],
      "before_update": Array [],
      "bound": Object {},
      "callbacks": Object {},
      "context": Map {},
      "ctx": Array [],
      "dirty": Array [
        -1,
      ],
      "fragment": Object {
        "c": [Function],
        "d": [Function],
        "i": [Function],
        "l": [Function],
        "m": [Function],
        "o": [Function],
        "p": [Function],
      },
      "not_equal": [Function],
      "on_destroy": Array [
        [Function],
      ],
      "on_mount": Array [],
      "props": Object {},
      "update": [Function],
    },
  },
  "container": <body>
    <div>
      <footer>
        <h2>
          Gamecheap 2020
        </h2>
      </footer>
    </div>
  </body>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
